app/Actions/FilterBooking.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Actions;
+
+use App\Models\Booking;
+use Carbon\Carbon;
+
+class FilterBooking
+{
+    public function execute($request)
+    {
+        $bookings = Booking::where('client_id', '=', $request['client_id'])->get();
+
+        if ($request['query'] == 'future') {
+            return $bookings->where('start', '>', Carbon::now());
+        } elseif ($request['query'] == 'past') {
+            return $bookings->where('start', '<', Carbon::now());
+        } else {
+            return $bookings;
+        }
```

Comment:
It's not very efficient, because you first `->get()` all the bookings for the client, which means these models are loaded into memory, and only then apply the filter for either future or past bookings.

It's much more efficient to apply these filters on the query level, e.g.:
```suggestion
        $bookings = Booking::where('client_id', '=', $request['client_id']);

        if ($request['query'] == 'future') {
            $bookings = $bookings->where('start', '>', Carbon::now());
        } elseif ($request['query'] == 'past') {
            $bookings = $bookings->where('start', '<', Carbon::now());
        }
        
        return $bookings->get();
```